app/controllers/modules/course/announcements_module.rb

```diff
@@ -5,7 +5,10 @@ class Course::AnnouncementsModule
     [
       {
         title: I18n.t('course.announcements.sidebar_title'),
-        unread: 0
+        unread:
+          if @course && @current_user
+            @course.announcements.unread_by(@current_user).count
```

Comment:
only way to have the general function is to use symbols and send, which I think is nasty.

move this if block to another function in the module. that function can be refactored out if we think it is necessary in future.